save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/Action.kt

```diff
@@ -0,0 +1,151 @@
+
+val actionButton: FC<ActionProps> = FC { props ->
+    val windowOpenness = useWindowOpenness()
+    val (displayTitle, setDisplayTitle) = useState(props.title)
+    val (displayMessage, setDisplayMessage) = useState(props.message)
+    val (isError, setError) = useState(false)
+    val (isClickMode, setClickMode) = useState(false)
+
+    val action = useDeferredRequest {
+        val response = props.sendRequest(isClickMode)(this)
+        if (response.ok) {
+            props.onActionSuccess(isClickMode)
+        } else {
+            setDisplayTitle(props.errorTitle)
+            setDisplayMessage(response.unpackMessage())
+            setError(true)
+            windowOpenness.openWindow()
+        }
+    }
+
+    div {
+        button {
+            type = ButtonType.button
+            className = ClassName(props.classes)
+            props.buttonStyleBuilder(this)
+            onClick = {
+                setDisplayTitle(props.title)
+                setDisplayMessage(props.message)
+                windowOpenness.openWindow()
+            }
+        }
+    }
+
+    displayModalWithClick(
+        title = displayTitle,
+        message = displayMessage,
+        isOpen = windowOpenness.isOpen(),
+        onCloseButtonPressed = windowOpenness.closeWindowAction(),
+        buttonBuilder = {
+            if (isError) {
+                buttonBuilder("Ok") {
+                    windowOpenness.closeWindow()
+                    setError(false)
+                }
+            } else {
+                props.modalButtons(action, windowOpenness, this)
+            }
+        },
+        clickBuilder = {
+            if (props.conditionClick && !isError) {
+                div {
+                    className = ClassName("d-sm-flex justify-content-center form-check")
+                    div {
+                        className = ClassName("d-sm-flex justify-content-center form-check")
+                        div {
+                            ReactHTML.input {
+                                className = ClassName("click")
+                                type = InputType.checkbox
+                                value = isClickMode
+                                id = "click"
+                                checked = isClickMode
+                                onChange = {
+                                    setClickMode(!isClickMode)
+                                }
+                            }
+                        }
+                        div {
+                            ReactHTML.label {
+                                className = ClassName("click")
+                                htmlFor = "click"
+                                +props.clickMessage
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    )
+}
+
+/**
+ * Button with modal for some action
+ *
+ * @return noting
+ */
+external interface ActionProps : Props {
```

Comment:
ActionProps is not clear - ButtonWithActionProps should be better imo

Reply:
Done